### ISSUE TITLE
[Question] ValueCAN 4-2 Network IDs

### ISSUE THREAD

**Issue by wreighsantos** [intrepidcs/icsneoapi#3]
Hi!

Since your documentation is not yet updated, I can't find any reference on what Network IDs to use for the FD channels of ValueCAN 4-2

I know for sure that the CAN1 and CAN2 are HSCAN and MSCAN since those are used for ValueCAN3, but for ValueCAN 4-2, I can't decipher what Network ID names to use for the 2 FD channels.

Can you please help?

Thank you!

**Comment by sangeetha-ponnusamy** [comment]
Hello!

Setting the icsSpyMessage.Protocol field to SPY_PROTOCOL_CANFD will treat the message on HS/MS as CANFD.

#define SPY_PROTOCOL_CANFD 30 (Treat it as CANFD)
#define SPY_PROTOCOL_CAN 1 (Treat it as CAN)

**Comment by sangeetha-ponnusamy** [comment]
Also, there is a new API "icsneoTxMessagesEx" which needs to be added here in this source for CANFD purpose, We will update the source soon.

**Comment by wreighsantos** [comment]
hi! thank you very much for your reply!

So here are my assumptions.
If protocol is set to CAN, it send it to CAN channel, otherwise, it will be sent to CAN FD channel?

So my last question if, if I want to send a CAN message on CAN FD channel, how would I do it?

**Comment by sangeetha-ponnusamy** [comment]
Hi,

Both channels will support both CAN and CANFD, Hence you can send CAN and CANFD messages on both channels using the protocol field.

**Comment by wreighsantos** [comment]
Hello again, sorry for bothering. I have a question, I'm really sorry as I don't have anyone to ask to :-( Pardon my ignorance.

Is the settings struct "SVCAN412Settings" used for ValueCAN4-2?

**Comment by wreighsantos** [comment]
Thank you! I promise that would be my last. :smile:

**Comment by wreighsantos** [comment]
Hi @sangeetha-ponnusamy, sorry for bothering you again, will it be possible for you to answer my question in https://github.com/intrepidcs/icsneoapi/issues/6? Thank you!

**Comment by sangeetha-ponnusamy** [comment]
Hello, Sorry about the delay.

Please use this below check to retrieve the ESI bit.

Msg.StatusBitField3 & SPY_STATUS3_CANFD_ESI should be true if ESI bit is set.

**Comment by wreighsantos** [comment]
Thank you!

**Comment by wreighsantos** [comment]
Hi @sangeetha-ponnusamy, I have a question
Sorry for bugging you again, I can't find any reference, even in Intrepid's latest documentation.
I just recently bought a ValueCAN4-2 device and it didn't come with an updated documentation and helper files.

Right now, I'm trying to send a CAN-FD message with the BRS switch on.
So I set StatusbitField3 to 16, then, it always end up as a frame with error.
I even tried using Vehicle Spy to send but it always says Tx Aborted.

Are there any additional settings that I need to configure to make it work?
Because I saw in other devices, they have configurations for Arbitration Phase Baud Rate and Data Phase Baud Rate, however, I can't find the same thing in Neo Explorer. Does it cause the BRS enabled frame to end up having an error?

I'm really sorry for the inconvenience that it may cause you answering my questions, I appreciate you entertaining my questions.

Thank you!
